src/main.cpp

```diff
@@ -3238,7 +3238,11 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
             sapling_tree.append(outputDescription.cmu);
         }
 
-        orchard_tree.AppendBundle(tx.GetOrchardBundle());
+        if (!orchard_tree.AppendBundle(tx.GetOrchardBundle())) {
+            return state.DoS(100,
+                error("ConnectBlock(): the Orchard commitment tree is full."),
```

Comment:
Technically the tree is not necessarily full; it might still be possible for a block with fewer Orchard notes to get mined.
```suggestion
                error("ConnectBlock(): block would overfill the Orchard commitment tree."),
```